### PR TITLE
update DSL: cloud prov (aws, hp_cloud) content

### PR DESCRIPTION
**Pull request by Bonczidai**
Update cloud prov (aws, hp_cloud) content to the new expressions and string constants syntax 
Signed-off-by: Bonczidai Levente <levente.bonczidai@hpe.com>

### CODE REVIEW

File: content/io/cloudslang/cloud_provider/hp_cloud/net/add_ip_to_server.sl [hunk 1]
```diff
@@ -47,11 +47,11 @@ flow:
...
           rest.http_client_post:
-            - url: "'https://region-' + region + '.geo-1.compute.hpcloudsvc.com/v2/' + tenant + '/servers/' + server_id + '/action'"
-            - headers: "'X-AUTH-TOKEN:' + token"
-            - content_type: "'application/json'"
+            - url: ${'https://region-' + region + '.geo-1.compute.hpcloudsvc.com/v2/' + tenant + '/servers/' + server_id + '/action'}
+            - headers: ${'X-AUTH-TOKEN:' + token}
+            - content_type: 'application/json'
             - body: >
-                '{"addFloatingIp": { "address": "'+ip_address+'" }}'
+                ${'{"addFloatingIp": { "address": "'+ip_address+'" }}'}
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/net/add_ip_to_server.sl]: Shouldn't it be without the quotes around the expr?
`${ {"addFloatingIp": { "address": "'+ip_address+'" }} }`

File: content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip.sl [hunk 2]
```diff
@@ -43,11 +43,11 @@ flow:
...
         do:
           rest.http_client_post:
-            - url: "'https://region-' + region + '.geo-1.network.hpcloudsvc.com/v2.0/floatingips'"
-            - headers: "'X-AUTH-TOKEN:' + token"
+            - url: ${'https://region-' + region + '.geo-1.network.hpcloudsvc.com/v2.0/floatingips'}
+            - headers: ${'X-AUTH-TOKEN:' + token}
             - body: >
-                '{"floatingip": { "floating_network_id": "' + ext_network_id + '" }}'
-            - content_type: "'application/json'"
+                ${'{"floatingip": { "floating_network_id": "' + ext_network_id + '" }}'}
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip.sl]: Shouldn't it be something like:
` ${ "{ floatingip: { floating_network_id: " + ext_network_id + " }}" } ` ?

File: content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip_flow.sl [hunk 3]
```diff
@@ -64,24 +64,24 @@ flow:
...
           - status_code
 
     - get_ip_result:
         do:
-          json.get_value_from_json:
-            - json_input: return_result
-            - key_list: ["'floatingip'", "'floating_ip_address'"]
+          json.get_value:
+            - json_input: ${return_result}
+            - json_path: ["'floatingip'", "'floating_ip_address'"]
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip_flow.sl]: Why do we still need the double quotes on the values?
Can't it be: ` - json_path: ["floatingip", "floating_ip_address"] ` ?

File: content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip_flow.sl [hunk 4]
```diff
@@ -64,24 +64,24 @@ flow:
...
         publish:
-          - ip_address: value
+          - ip_address: ${value}
 
     - on_failure:
       - ip_error:
           do:
             print.print_text:
-              - text: "'! ERROR ALLOCATING IP (' + status_code + ') : ' + return_result" 
+              - text: "${'! ERROR ALLOCATING IP (' + status_code + ') : ' + return_result}"
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/net/create_floating_ip_flow.sl]: Shouldn't it be without the wrapping double quotes:
` - text: ${ '! ERROR ALLOCATING IP (' + status_code + ') : ' + return_result} ` ?

File: content/io/cloudslang/cloud_provider/hp_cloud/create_server_flow.sl [hunk 5]
```diff
@@ -163,20 +163,20 @@ flow:
...
-            - proxy_port         
+            - proxy_port
         publish:
           - return_result
 
     - done:
         do:
           print.print_text:
-            - text: "'### New server (' + server_name + ') is ready'"
+            - text: "${'### New server (' + server_name + ') is ready'}"
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/create_server_flow.sl]: Are the wrapping double quotes needed?

File: content/io/cloudslang/cloud_provider/hp_cloud/create_server_flow.sl [hunk 6]
```diff
@@ -163,20 +163,20 @@ flow:
...
 
     - on_failure:
       - create_server_error:
           do:
             print.print_text:
-              - text: "'! Create Server Flow Error: ' + return_result" 
+              - text: "${'! Create Server Flow Error: ' + return_result}"
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/create_server_flow.sl]: Are the wrapping double quotes needed?

File: content/io/cloudslang/cloud_provider/hp_cloud/delete_server_flow.sl [hunk 7]
```diff
@@ -100,7 +100,7 @@ flow:
     - done:
         do:
           print.print_text:
-            - text: "'### Server ('+server_id+') was removed'"
+            - text: "${'### Server ('+server_id+') was removed'}"
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/delete_server_flow.sl]: Again with the double quotes

File: content/io/cloudslang/cloud_provider/hp_cloud/get_server_state_flow.sl [hunk 8]
```diff
@@ -92,10 +92,10 @@ flow:
     - failed:
           do:
             print.print_text:
-              - text: "'! ERROR GETTING SERVER INFO: \\nStatus:' + status_code + '\\n' + return_result"
+              - text: "${'! ERROR GETTING SERVER INFO: \\nStatus:' + status_code + '\\n' + return_result}"
```
Comment (oritstone) [inline content/io/cloudslang/cloud_provider/hp_cloud/get_server_state_flow.sl]: Double quotes...

**Comment by oritstone** [comment]
@Bonczidai I made some small comments
Is it tested yet?

**Comment by Bonczidai** [comment]
in the remaining cases, the quotes are necessary

**Comment by oritstone** [comment]
:+1:

**Pull request merged**